app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb

```diff
@@ -0,0 +1,279 @@
+module Hyrax
+  module Dashboard
+    module CollectionsControllerBehavior
```

Comment:
I think this would make the refactor much less complex as you'd move code to an existing file rather than creating a new module. We aren't using prepend in Hyrax itself, but you can use it to inject behavior in your local app.  It's a core language feature available since ruby 2.0 http://ruby-doc.org/core-2.0.0/Module.html#method-i-prepend